retry_delay.go

```diff
@@ -53,9 +52,14 @@ func newDefaultBackoff(maxDelay time.Duration) backoffStrategy {
 }
 
 func (s defaultBackoffStrategy) applyBackoff(baseDelay time.Duration, retryCount int) time.Duration {
-	d := time.Duration(int64(baseDelay) * int64(math.Pow(2, float64(retryCount))))
```

Comment:
Is there a reason we can't just do the conversion and then check for overflow?

Reply:
What would that look like, specifically? Are you suggesting that we compare the result of `math.Pow` (as a float64) to 2^63— or try to determine if there was an overflow after we have converted to `int64`?

Reply:
Maybe something like `Math.min(float64(baseDelay) * math.Pow(2, float64(retryCount)), float64(s.maxDelay))`?

Reply:
That's reasonable. I'm not sure what I was thinking there— I had considered that, and then I had some concern about loss of precision but I can't reconstruct my train of thought and I think it must have been a mistake.